Tickets/dm 31770

pcortes80 opened the pull request:
- Replace the use of user events by occurrence to stop the servers.
- Added stopServer occurrence timeout  into the configurations.
- Decided to keep the Event Structure in **TcpServerTel.runServer.vi** to be consistent with **TcpServerCmd.runServer.vi** and to avoid add more complexity to the code.
- Added runServer timeout into the configurations.

teweitsai reviewed (approved):
[Review of ts_labview_tcp_ip]

[ComponentSimulator.vi]

1. Same question of TcpServer.configServer.vi.


<<TcpServer.lvclass>>

[configServer.vi]

1. For the timeout of "runServerTimeoutInMs" and "stopServerTimeoutInMs", why not reuse the "serverTimeout (ms)" in the configuration in the front panel? This should simply the configuration. Is there any reason for these two to be different?


<<TcpServerBase.lvclass>>

1. I think there are too many timeout attributes right now in the class. I think we could decrease the number of them (by the reusing of the same timeout value).

[Write_stopServerTimeoutInMs.vi]

1. There is the overlappting in block diagram. This is not good. Please help to fix it.

[Read_stopServerOcc.vi]

1. There is the overlappting in block diagram. This is not good. Please help to fix it.
2. For the description of "stopServerOcc", you assumed the parent has the knowledge of child. This is not good. The idea of OOD is the hierarchy and the parent should have no idea of child.

[Write_stopServerOcc.vi]

1. There is the overlappting in block diagram. This is not good. Please help to fix it.
2. For the description of "stopServerOcc", you assumed the parent has the knowledge of child. This is not good. The idea of OOD is the hierarchy and the parent should have no idea of child.

[regStopServerOcc.vi]

1. For the description of "regStopServerOcc", you assumed the parent has the knowledge of child. This is not good. The idea of OOD is the hierarchy and the parent should have no idea of child.

2. I am think that not use the "Preallocated reentrant clone execution" for the reuse of code for this vi. Let us discuss this.



<<TcpServerCmd.lvclass>>

[setStopServerOcc.vi]

1. Can we put this function to TcpServerBase.lvclass? I felt we could do this. This is a duplication of TcpServerTel.setStopServerOcc.vi.

[waitStopServerOcc.vi]

1. Same question of setStopServerOcc.vi.


<<TcpServerTel.lvclass>>

[setStopServerOcc.vi]

1. Can we put this function to TcpServerBase.lvclass? I felt we could do this. This is a duplication of TcpServerCmd.setStopServerOcc.vi.

[waitStopServerOcc.vi]

1. Same question of setStopServerOcc.vi.

[runServer.vi]

1. Do we need the event case structure now?


[testStopServer.vi]

1. The message in block diagram can not differentiate which server. Please correct it.

pcortes80 commented:
[Replies of ts_labview_tcp_ip]

[ComponentSimulator.vi]

Same question of TcpServer.configServer.vi.

Reply:
- Solve in the following point.

<<TcpServer.lvclass>>

[configServer.vi]

For the timeout of "runServerTimeoutInMs" and "stopServerTimeoutInMs", why not reuse the "serverTimeout (ms)" in the configuration in the front panel? This should simply the configuration. Is there any reason for these two to be different?

Reply:
- As we discussed yesterday, I added them to do the code more flexibly, but I take your advice of reusing the “serverTimeout (ms)” to avoid confusing the User.

<<TcpServerBase.lvclass>>

I think there are too many timeout attributes right now in the class. I think we could decrease the number of them (by reusing the same timeout value).

Reply:
- I removed the extra timeouts to reuse the “serverTimeout (ms)”.

[Write_stopServerTimeoutInMs.vi]

There is the overlappting in block diagram. This is not good. Please help to fix it.

Reply:
- VI removed.

[Read_stopServerOcc.vi]

There is overlapping in the block diagram. This is not good. Please help to fix it.

Reply:
- Fixed.

For the description of "stopServerOcc", you assumed the parent has the knowledge of child. This is not good. The idea of OOD is the hierarchy and the parent should have no idea of child.

Reply:
- Description fixed.

[Write_stopServerOcc.vi]

There is the overlapping in the block diagram. This is not good. Please help to fix it.

Reply:
- Fixed.

For the description of "stopServerOcc", you assumed the parent has the knowledge of child. This is not good. The idea of OOD is the hierarchy and the parent should have no idea of child.

Reply:
- Description fixed.

[regStopServerOcc.vi]

For the description of "regStopServerOcc", you assumed the parent has the knowledge of child. This is not good. The idea of OOD is the hierarchy and the parent should have no idea of child.

Reply:
- Description fixed.

I think that not use the "Preallocated reentrant clone execution" for the reuse of code for this vi. Let us discuss this.

Reply:
- Changed the "Preallocated reentrant clone execution" option to “Non-reentrant execution”.


<<TcpServerCmd.lvclass>>

[setStopServerOcc.vi]

Can we put this function to TcpServerBase.lvclass? I felt we could do this. This is a duplication of TcpServerTel.setStopServerOcc.vi.

Reply:
- Add a new function called TcpServerBase.setStopServerOcc.vi.
- Rename the TcpServerCmd.setStopServerOcc.vi to TcpServerCmd.triggertStopServerOcc.vi.

[waitStopServerOcc.vi]

Same question of setStopServerOcc.vi.

Reply:
- I decided not to move this function into de Base, because of encapsulation. 

<<TcpServerTel.lvclass>>

[setStopServerOcc.vi]

Can we put this function to TcpServerBase.lvclass? I felt we could do this. This is a duplication of TcpServerCmd.setStopServerOcc.vi.

Reply:
- Removed the TcpServerTel.setStopServerOcc.vi.

[waitStopServerOcc.vi]

Same question of setStopServerOcc.vi.

Reply:
- I decided not to move this function into de Base, because of encapsulation. 

[runServer.vi]

Do we need the event case structure now?

Reply:
- Yes, we do. I decided to keep the Event Structure to be consistent with TcpServerCmd.runServer.vi code style, and if in the future we want to register a new dynamic user event, we can register it, because we already have the event structure.
- I put a label in the code to explain this.

[testStopServer.vi]

The message in block diagram can not differentiate which server. Please correct it.

Reply:
- Fixed.
